File: Glamify.Portal/src/model/requests/CreateAppointmentRequest.ts
import { AppointmentStatus } from "../Appointment"
import { Business } from "../Business"
import { Employee } from "../Employee"
import { Service } from "../Service"
import { User } from "../User"

export interface CreateAppointmentRequest {
  userId: number
  user: User
  locationId: number
  location: Business
  appointmentTime: Date
  status: AppointmentStatus
  services: Service[]
  employeeId?: number
  employee?: Employee
}

File: Glamify.Server/DTOs/AppointmentDTOs/CreateAppointmentRequest.cs
using Glamify.Domain.Entities;
using Glamify.Domain.Enums;

namespace Glamify.API.DTOs.AppointmentDTOs
{
    public class CreateAppointmentRequest
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public int LocationId { get; set; }
        public Business Location { get; set; } = null!;
        public DateTime AppointmentTime { get; set; }
        public AppointmentStatus Status { get; set; }
        public virtual ICollection<Service> Services { get; set; } = new List<Service>();
    }
}
